src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,78 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+interface Props {
+  actionClose: () => void,
+  selectedTodo: Todo,
+  changeSelectedTodo: (el: Todo | null) => void,
+}
+
+export const TodoModal: React.FC<Props> = ({
+  actionClose,
+  selectedTodo,
+  changeSelectedTodo,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId)
+      .then(res => setSelectedUser(res));
+  }, [selectedTodo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {selectedUser
+        ? (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${selectedTodo?.id}`}
```

Comment:
selectedTodo cannot be undefined according to your props interface
<img width="244" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/70220667/9e71778e-66ea-4664-a154-744a1646de29">
```suggestion
                {`Todo #${selectedTodo.id}`}
```